src/runtime/cuda.cpp

```diff
@@ -778,22 +778,41 @@ WEAK int halide_cuda_buffer_copy(void *user_context, struct halide_buffer_t *src
     return err;
 }
 
+namespace {
+
+WEAK int cuda_device_offset(const struct halide_buffer_t *src,
```

Comment:
Reading just the prototype here, I'd assume this routine returns an offset.

Reply:
Good point. Updated all the new helper functions to better names,